### DISCUSSION TITLE
Robots.txt?

### DISCUSSION THREAD

**Discussion by maximjsx** [outline/outline#7247] [Self Hosting] [answered]
I want to make my (selfhosted) outline instance publically accessible... Problem is that the site cannot be crawled by google search.
Looks like the robots.txt is hardcoded (https://github.com/outline/outline/blob/6a6c069e4e9f737edf42b5fdcdf4fccd640023b6/server/utils/robots.ts)

So thats what is returned from https://domain/robots.txt:
```
User-agent: *
Disallow: /
```

Is there any way to change that without building outline myself from source?

**Comment by maximjsx** [comment]
Can I just set isCloudHosted in the .env to true?

**Reply by tommoor** [reply]
It's not possible to make an Outline instance publicly accessible, the platform is designed for private documentation behind a login

**Comment by maximjsx** [comment]
Or does it break other things

**Comment by maximjsx** [comment]
nevermind its not an environment variable :/
![image](https://github.com/user-attachments/assets/e1807367-f039-434c-9e28-7e44a933f55b)

**Reply by tommoor** [reply]
This is by design to prevent misconfiguration, it would break many things to change this in a self-hosted installation

**Comment by maximjsx** [comment]
So It's not possible to list it on google search?

**Reply by AndrassGray** [reply]
Can't you simply define a specific endpoint and serve a local file ?

**Reply by maximjsx** [reply]
> Can't you simply define a specific endpoint and serve a local file ?

Where should I define it?

**Reply by AndrassGray** [reply]
Within your nginx configuration.

`    location /robots.txt {
            index my_robots.txt;
    }
`

**Reply by maximjsx** [reply]
> Within your nginx configuration.
> 
> ` location /robots.txt { index my_robots.txt; }`

I'm using the docker setup with the built in https-portal method...
Do you know where the nginx configuration for that is located and in which container?

**Reply by AndrassGray** [reply]
No idea, I don't know anything about https-portal.

But from their documentation : 

> 
> Override Nginx Configuration Files
> 
> You can override default nginx settings by providing a config segment of nginx.conf containing a valid server block. The custom nginx configurations are [ERB](http://www.stuartellis.eu/articles/erb/) templates and will be rendered before usage.
> 
> You can either just override one single site's config or all sites' configs.

So I guess it should be possible. Maybe knock doors at https://github.com/SteveLTN/https-portal ?

**Reply by AndrassGray** [reply, marked as answer]
Checkout [this section of the https-portal doc](https://github.com/SteveLTN/https-portal?tab=readme-ov-file#other-server-block-level-configurations).

**Reply by maximjsx** [reply]
Yea looks like it works now! Thanks.
![image](https://github.com/user-attachments/assets/c9d057e2-61d1-4e11-839d-c314cff20ce8)

**Reply by maximjsx** [reply]
![image](https://github.com/user-attachments/assets/4e15527c-c45d-44cf-8839-f0cba3831860)

**Reply by AndrassGray** [reply]
Happy you got this sorted. Well done!
